### PR TITLE
Allow to use Doctrine MongoDB registry if default Doctrine is not available

### LINKED ISSUE #2201: [ResourceBundle] Unresolved dependency using mongodb
Using MongoDB as ODM, the service `doctrine` doesn't exist. The service `sylius.form.type.entity_hidden` has a dependency on it:

File: src/Sylius/Bundle/ResourceBundle/Resources/config/services.xml

```xml
<service id="sylius.form.type.entity_hidden" class="%sylius.form.type.entity_hidden.class%">
    <argument type="service" id="doctrine" />
    <tag name="form.type" alias="entity_hidden" />
</service>
```

The dependency should be `doctrine_mongodb` if MongoDB is used. You can use `expression language` with Symfony 2.4:

```xml
<argument type="expression">container.hasService('doctrine_mongodb') ? service('doctrine_mongodb') : service('doctrine')</argument>
```

### PR DESCRIPTION
**Pull request by stloyd**
Should fix #2201.

### CODE REVIEW

**Comment by stloyd** [comment]
@pjedrzejewski @aRn0D @rubenrua Feedback?

**Comment by rubenrua** [comment]
:+1:

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/ObjectToIdentifierServicePass.php [hunk 1]
```diff
@@ -46,6 +49,11 @@ public function process(ContainerBuilder $container)
             if (!$container->hasDefinition('sylius_entity_to_identifier')) {
                 $container->setAlias('sylius_entity_to_identifier', 'sylius_document_to_identifier');
             }
+
+            if (!$container->hasDefinition('doctrine')) {
+                $definition = $container->findDefinition('sylius.form.type.entity_hidden');
+                $definition->replaceArgument(0, new Reference('doctrine_mongodb'));
```
Comment (adamelso) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/ObjectToIdentifierServicePass.php]: Would the service not be renamed from `sylius.form.type.entity_hidden` to `sylius.form.type.model_hidden` ?

Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/ObjectToIdentifierServicePass.php]: Personally I start to prefer smaller PRs as @pjedrzejewski is not having time to review & we wait way to long for merges, so this can be changed in another PR IMO.

Comment (adamelso) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/ObjectToIdentifierServicePass.php]: In that case, I'm :+1:

**Comment by stloyd** [comment]
@pjedrzejewski ping...
